src/App.tsx

```diff
@@ -1,14 +1,56 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useContext, useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { TodosContext } from './context/TodosContext';
+import { Select } from './types/Select';
+import { useDebounce } from './hooks/useDebounce';
 
 export const App: React.FC = () => {
+  const { show, filterField, query } = useContext(TodosContext);
+
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const debouncedQuery = useDebounce(query);
+
+  useEffect(() => {
+    getTodos().then(todosFromSetver => setTodos(todosFromSetver));
+  });
+
+  function prepareTodos(listOfTodos: Todo[]) {
+    let preparedTodos = [...listOfTodos];
+
+    preparedTodos = preparedTodos.filter(todo => {
+      switch (filterField) {
+        case Select.Active:
+          return !todo.completed;
+
+        case Select.Completed:
+          return todo.completed;
+
+        default:
+          return todo;
+      }
+    });
+
+    if (debouncedQuery) {
+      const queryNormalize = debouncedQuery.toLowerCase();
+
+      preparedTodos = preparedTodos
+        .filter(todo => todo.title.toLowerCase().includes(queryNormalize));
+    }
+
+    return preparedTodos;
+  }
```

Comment:
you can move it outside component file.
If you can write constant instead let use it prefer.
let worsens the readability and perception of the code during scaling